README.adoc

```diff
@@ -692,3 +692,69 @@ The color is red and the shape is square.
 ----
 
 --
+
+
+=== Content contains json or yaml filepaths
+
+When the content of the loaded files contains filepaths of json files or
+yaml files, it will load extra data by these filepaths.  The load content
+becomes a hash of `file-data` pairs.
+
+Where:
+
+* `file` is the filepath
+* `data` is the content of the file
+
+EXAMPLE:
+--
+Given:
+
+strings1.json
+[source,json]
+----
+{
+  "foo": "bar",
+  "dead": ["a.yaml", "b.yaml"]
+}
+----
+
+a.yaml
+[source,yaml]
+----
+---
+shape: circle
+color: red
+----
+
+b.yaml
+[source,yaml]
+----
+---
+shape: square
+color: blue
+corners: 4
+----
+
+And the block:
+[source,asciidoc]
+------
+[data2text,my_context=strings1.json]
+----
+I'm heading to the {{my_context.foo}}.
+
+{% for item in my_context.dead %}
+This is {{ item['data'][shape] }} with color {{ item['data'][color] }}.
```

Comment:
@kwkwan instead of using `['data']` to trigger loading the file, how about we define a custom filter such as `load_file` to load the file? This way we can use `{% assign foo = load_file: item %}`.